packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx

```diff
@@ -91,14 +95,63 @@ export const ImpersonationFab = () => {
   const session = useCoreSession();
   const { t } = useLocalizations();
   const { parsedInternalTheme } = useAppearance();
+  const ref = useRef<HTMLDivElement>(null);
   const actor = session?.actor;
   const isImpersonating = !!actor;
 
   //essentials for calcs
-  const top = '109px';
-  const right = '23px';
   const eyeWidth = parsedInternalTheme.sizes.$16;
   const eyeHeight = eyeWidth;
+  const topProperty = '--cl-impersonation-fab-top';
+  const rightProperty = '--cl-impersonation-fab-right';
+  const defaultTop = 109;
+  const defaultRight = 23;
+
+  const handleResize = () => {
+    const { current } = ref;
+    if (!current) {
+      return;
+    }
+
+    if (current.offsetLeft < 0 || current.offsetLeft > window.innerWidth) {
+      document.documentElement.style.setProperty(rightProperty, `${defaultRight}px`);
+    }
+  };
+
+  const onMouseDown: MouseEventHandler = () => {
+    window.addEventListener('mousemove', onMouseMove);
+    window.addEventListener(
+      'mouseup',
+      () => {
+        window.removeEventListener('mousemove', onMouseMove);
+        handleResize();
+      },
+      { once: true },
+    );
+  };
+
+  const onMouseMove = React.useCallback((e: MouseEvent) => {
+    e.stopPropagation();
+    e.preventDefault();
+    const { current } = ref;
+    if (!current) {
+      return;
+    }
+    document.documentElement.style.setProperty(
+      rightProperty,
+      `${window.innerWidth - current.offsetLeft - current.offsetWidth - e.movementX}px`,
```

Comment:
🙃 let's extract this magic expression into a new variable with a descriptive name to better document its intent even if it ends up being a rather long one, .e.g.:

```
const rightOffestBasedOnViewportAndContent = `${window.innerWidth - current.offsetLeft - current.offsetWidth - e.movementX}px`;
document.documentElement.style.setProperty(
      rightProperty,
      rightOffestBasedOnViewportAndContent,
```

Feel free to suggest better names, of course!